src/main/java/ghidrathon/interpreter/GhidrathonInterpreter.java

```diff
@@ -139,28 +295,33 @@ public void write(byte[] b, int off, int len) throws IOException {
    * @throws JepException
    * @throws FileNotFoundException
    */
-  private void setJepNativeBinaryPath() throws JepException, FileNotFoundException {
-
-    File nativeJep;
-
-    try {
-
-      nativeJep = Application.getOSFile(GhidrathonUtils.THIS_EXTENSION_NAME, "libjep.so");
-
-    } catch (FileNotFoundException e) {
-
-      // whoops try Windows
-      nativeJep = Application.getOSFile(GhidrathonUtils.THIS_EXTENSION_NAME, "jep.dll");
+  private void setJepPaths() throws JepException, FileNotFoundException {
+    // if this is set, it take precedence over VIRTUAL_ENV.
+    Path pythonPathJep = findPythonPathJep();
+    if (pythonPathJep != null) {
+      log.info("found JEP dll via PYTHONPATH: " + pythonPathJep);
     }
 
-    try {
+    // if this is set, it takes precedence over system python
+    Path virtualenvJep = findVirtualEnvJep();
+    if (virtualenvJep != null) {
+      log.info("found JEP dll via VIRTUAL_ENV: " + virtualenvJep);
+    }
 
-      MainInterpreter.setJepLibraryPath(nativeJep.getAbsolutePath());
+    // fall back to whatever python3 references
+    Path systemJep = findSystemJep();
+    if (systemJep != null) {
+        log.info("found JEP dll via python3: " + systemJep);
+    }
```

Comment:
we collect all the paths first so that we can log out the status fully, then we actually set the config second.